PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java

```diff
@@ -0,0 +1,249 @@
+package org.prebid.mobile;
+
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.prebid.mobile.testutils.BaseSetup;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.util.ArrayList;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertNull;
+import static junit.framework.Assert.assertTrue;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk = BaseSetup.testSDK, manifest=Config.NONE)
+public class NativeAdUnitTest {
+    static final String PBS_CONFIG_ID_NATIVE_APPNEXUS = "1f85e687-b45f-4649-a4d5-65f74f2ede8e";
+
+    @Test
+    public void testNativeAdUnitCreation() throws Exception {
+        NativeAdUnit nativeUnit = new NativeAdUnit(PBS_CONFIG_ID_NATIVE_APPNEXUS);
+        assertNotNull(nativeUnit);
+        assertEquals(PBS_CONFIG_ID_NATIVE_APPNEXUS, FieldUtils.readField(nativeUnit, "configId", true));
+        assertEquals(AdType.NATIVE, FieldUtils.readField(nativeUnit, "adType", true));
+    }
+
+    @Test
+    public void testNativeAdContextType() {
+        NativeAdUnit nativeUnit = new NativeAdUnit(PBS_CONFIG_ID_NATIVE_APPNEXUS);
+        assertNotNull(nativeUnit);
```

Comment:
Not only check returned type is context type, also check the id fo the type is correct value.

Reply:
Done